Windows CI not succeeding

## Description
Latest `main` does not pass CI on Windows
 - This is follow up from #62 
See example build:
https://ci.ros2.org/job/ci_windows/18107/console

### Expected Behavior
Windows CI shall pass green without errors and warnings

## Additional context
Build fails with error message:
```
CMake Error at C:/ci/ws/install/share/mcap_vendor/cmake/mcapExport.cmake:85 (message):
16:13:36 CMake Error at C:/ci/ws/install/share/mcap_vendor/cmake/mcapExport.cmake:85 (message):
16:13:36   The imported target "mcap_vendor::mcap" references the file
16:13:36      "C:/ci/ws/install/lib/mcap.lib"
16:13:36   but this file does not exist.  Possible reasons include:
16:13:36   * The file was deleted, renamed, or moved to another location.
16:13:36   * An install or uninstall procedure did not complete successfully.
16:13:36   * The installation package was faulty and contained
16:13:36      "C:/ci/ws/install/share/mcap_vendor/cmake/mcapExport.cmake"
16:13:36   but not all the files it references.
16:13:36 Call Stack (most recent call first):
16:13:36   C:/ci/ws/install/share/mcap_vendor/cmake/ament_cmake_export_targets-extras.cmake:9 (include)
16:13:36   C:/ci/ws/install/share/mcap_vendor/cmake/mcap_vendorConfig.cmake:41 (include)
16:13:36   CMakeLists.txt:23 (find_package)
```
From the log file It looks like `mcap.lib` hasn't been built during the build process for some unknown reasons.
Also from logs it looks like it was built `mcap.dll` but not `mcap.lib`.

@amacneil We also have bunch of warnings on Windows build in `mcap` sources which we need to address.
```
Checking Build System
  Building Custom Rule C:/ci/ws/src/ros-tooling/rosbag2_storage_mcap/mcap_vendor/CMakeLists.txt
  main.cpp
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\internal.hpp(54,46): warning C4267: '+=': conversion from 'size_t' to 'uint32_t', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\reader.inl(67,23): warning C4244: 'argument': conversion from 'uint64_t' to 'long', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\reader.inl(324,16): warning C4996: 'fopen': This function or variable may be unsafe. Consider using fopen_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\reader.inl(548,43): warning C4267: '=': conversion from 'size_t' to 'uint16_t', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\reader.inl(549,45): warning C4267: '=': conversion from 'size_t' to 'uint32_t', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\reader.inl(550,57): warning C4267: '=': conversion from 'size_t' to 'uint32_t', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\reader.inl(551,53): warning C4267: '=': conversion from 'size_t' to 'uint32_t', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\reader.inl(552,47): warning C4267: '=': conversion from 'size_t' to 'uint32_t', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\reader.inl(1013,40): warning C4267: '+=': conversion from 'size_t' to 'uint32_t', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\reader.inl(1020,47): warning C4267: '+=': conversion from 'size_t' to 'uint32_t', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\reader.inl(1034,33): warning C4244: '+=': conversion from 'uint64_t' to 'uint32_t', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\reader.inl(1068,45): warning C4267: '+=': conversion from 'size_t' to 'uint32_t', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\crc32.hpp(86,44): warning C4267: 'initializing': conversion from 'size_t' to 'uint32_t', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\writer.inl(45,16): warning C4996: 'fopen': This function or variable may be unsafe. Consider using fopen_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\writer.inl(268,93): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\writer.inl(268,93): warning C4267: 'initializing': conversion from 'size_t' to 'const int', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\writer.inl(907,57): warning C4267: 'initializing': conversion from 'size_t' to 'uint32_t', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\writer.inl(907,57): warning C4267: 'initializing': conversion from 'size_t' to 'const uint32_t', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\writer.inl(924,81): warning C4267: 'initializing': conversion from 'size_t' to 'uint32_t', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\writer.inl(924,81): warning C4267: 'initializing': conversion from 'size_t' to 'const uint32_t', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\writer.inl(993,83): warning C4267: 'initializing': conversion from 'size_t' to 'uint32_t', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
C:\ci\ws\build\mcap_vendor\_deps\mcap-src\cpp\mcap\include\mcap\writer.inl(993,83): warning C4267: 'initializing': conversion from 'size_t' to 'const uint32_t', possible loss of data [C:\ci\ws\build\mcap_vendor\mcap.vcxproj]
  lz4.c
  lz4frame.c
  lz4hc.c
  xxhash.c
  mcap.vcxproj -> C:\ci\ws\build\mcap_vendor\Release\mcap.dll
```

--- a/mcap_vendor/CMakeLists.txt
+++ b/mcap_vendor/CMakeLists.txt
@@ -21,7 +21,8 @@ macro(build_mcap_vendor)
   include(FetchContent)
   fetchcontent_declare(mcap
     GIT_REPOSITORY https://github.com/foxglove/mcap.git
-    GIT_TAG 066f03b9b8444ffc1134e852e3419e4a4c0bef85 # releases/cpp/v0.6.0
+    # GIT_TAG 066f03b9b8444ffc1134e852e3419e4a4c0bef85 # releases/cpp/v0.6.0
+    GIT_TAG windows-build # releases
   )
   fetchcontent_makeavailable(mcap)
 
@@ -38,6 +39,9 @@ macro(build_mcap_vendor)
     src/main.cpp
     ${_lz4_srcs}
   )
+  # Causes the visibility macros to use dllexport rather than dllimport,
+  # which is appropriate when building the dll but not consuming it.
+  target_compile_definitions(mcap PRIVATE "MCAP_BUILDING_DLL")
 
   set(_mcap_include_dir ${mcap_SOURCE_DIR}/cpp/mcap/include)
 

--- a/mcap_vendor/CMakeLists.txt
+++ b/mcap_vendor/CMakeLists.txt
@@ -19,6 +19,8 @@ if (MSVC)
   add_compile_options(/W4 /WX
     /wd4251 # suppress warning about having template instances (such as std::string) as public class members
   )
+  # suppress error disallowing `fopen` in favor of `fopen_s`
+  add_compile_definitions(_CRT_SECURE_NO_WARNINGS)
 endif()
 
 ## Define vendor macro

--- a/mcap_vendor/CMakeLists.txt
+++ b/mcap_vendor/CMakeLists.txt
@@ -15,6 +15,12 @@ endif()
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic -Werror)
 endif()
+if (MSVC)
+  add_compile_options(/W4 /WX
+    # suppress warning about having template instances (such as std::string) as public class members
+    /wd4251
+  )
+endif()
 
 ## Define vendor macro
 macro(build_mcap_vendor)
@@ -34,14 +40,10 @@ macro(build_mcap_vendor)
 
   file(GLOB _lz4_srcs
     ${lz4_SOURCE_DIR}/lib/*.c)
-
   add_library(mcap SHARED
     src/main.cpp
     ${_lz4_srcs}
   )
-  # Causes the visibility macros to use dllexport rather than dllimport,
-  # which is appropriate when building the dll but not consuming it.
-  target_compile_definitions(mcap PRIVATE "MCAP_BUILDING_DLL")
 
   set(_mcap_include_dir ${mcap_SOURCE_DIR}/cpp/mcap/include)
 

--- a/mcap_vendor/CMakeLists.txt
+++ b/mcap_vendor/CMakeLists.txt
@@ -17,8 +17,7 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
 endif()
 if (MSVC)
   add_compile_options(/W4 /WX
-    # suppress warning about having template instances (such as std::string) as public class members
-    /wd4251
+    /wd4251 # suppress warning about having template instances (such as std::string) as public class members
   )
 endif()
 
@@ -27,8 +26,7 @@ macro(build_mcap_vendor)
   include(FetchContent)
   fetchcontent_declare(mcap
     GIT_REPOSITORY https://github.com/foxglove/mcap.git
-    # GIT_TAG 066f03b9b8444ffc1134e852e3419e4a4c0bef85 # releases/cpp/v0.6.0
-    GIT_TAG windows-build # releases
+    GIT_TAG dc6561d9ba867901709e36526dcf7f7359861e9c # releases/cpp/v0.7.0
   )
   fetchcontent_makeavailable(mcap)
 
@@ -40,6 +38,7 @@ macro(build_mcap_vendor)
 
   file(GLOB _lz4_srcs
     ${lz4_SOURCE_DIR}/lib/*.c)
+
   add_library(mcap SHARED
     src/main.cpp
     ${_lz4_srcs}

--- a/rosbag2_storage_mcap/.clang-format
+++ b/rosbag2_storage_mcap/.clang-format
@@ -17,6 +17,7 @@ FixNamespaceComments: true
 PointerAlignment: Left
 ReflowComments: true
 SortIncludes: true
+IndentPPDirectives: BeforeHash
 
 IncludeCategories:
   - Regex: '^"'

--- a/rosbag2_storage_mcap/CMakeLists.txt
+++ b/rosbag2_storage_mcap/CMakeLists.txt
@@ -34,6 +34,7 @@ target_include_directories(${PROJECT_NAME} PUBLIC
   $<INSTALL_INTERFACE:include/${PROJECT_NAME}>
 )
 target_compile_features(${PROJECT_NAME} PUBLIC c_std_99 cxx_std_17)
+target_compile_definitions(${PROJECT_NAME} PRIVATE "ROSBAG2_STORAGE_MCAP_BUILDING_DLL")
 ament_target_dependencies(${PROJECT_NAME}
   mcap_vendor
   pluginlib
@@ -88,6 +89,10 @@ if(BUILD_TESTING)
 
   set(ament_cmake_clang_format_CONFIG_FILE .clang-format)
   list(APPEND AMENT_LINT_AUTO_EXCLUDE ament_cmake_uncrustify)
+  if(WIN32)
+    # clang-format is not easily exposed to Windows, this linter enforced regardless via Linux/OSX builds
+    list(APPEND AMENT_LINT_AUTO_EXCLUDE ament_cmake_clang_format)
+  endif()
   ament_lint_auto_find_test_dependencies()
 
   ament_add_gmock(test_mcap_storage test/rosbag2_storage_mcap/test_mcap_storage.cpp)

--- a/rosbag2_storage_mcap/include/rosbag2_storage_mcap/message_definition_cache.hpp
+++ b/rosbag2_storage_mcap/include/rosbag2_storage_mcap/message_definition_cache.hpp
@@ -15,6 +15,8 @@
 #ifndef ROSBAG2_STORAGE_MCAP__MESSAGE_DEFINITION_CACHE_HPP_
 #define ROSBAG2_STORAGE_MCAP__MESSAGE_DEFINITION_CACHE_HPP_
 
+#include "visibility_control.hpp"
+
 #include <set>
 #include <string>
 #include <unordered_map>
@@ -67,6 +69,7 @@ public:
    * Throws DefinitionNotFoundError if one or more definition files are missing for the given
    * package resource name.
    */
+  ROSBAG2_STORAGE_MCAP_PUBLIC
   std::pair<Format, std::string> get_full_text(const std::string& package_resource_name);
 
 private:
@@ -87,6 +90,7 @@ private:
     msg_specs_by_definition_identifier_;
 };
 
+ROSBAG2_STORAGE_MCAP_PUBLIC
 std::set<std::string> parse_dependencies(Format format, const std::string& text,
                                          const std::string& package_context);
 

--- a/rosbag2_storage_mcap/include/rosbag2_storage_mcap/visibility_control.hpp
+++ b/rosbag2_storage_mcap/include/rosbag2_storage_mcap/visibility_control.hpp
@@ -12,11 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/* This header must be included by all rclcpp headers which declare symbols
- * which are defined in the rclcpp library. When not building the rclcpp
- * library, i.e. when using the headers in other package's code, the contents
- * of this header change the visibility of certain symbols which the rclcpp
- * library cannot have, but the consuming code must have inorder to link.
+/* This header must be included by this library's headers which declare symbols
+ * that are defined separately. The contents of this header change the visibility
+ * of certain symbols, especially for Windows DLL usage.
  */
 
 #ifndef ROSBAG2_STORAGE_MCAP__VISIBILITY_CONTROL_HPP_

--- a/rosbag2_storage_mcap/include/rosbag2_storage_mcap/visibility_control.hpp
+++ b/rosbag2_storage_mcap/include/rosbag2_storage_mcap/visibility_control.hpp
@@ -36,19 +36,14 @@
   #else
     #define ROSBAG2_STORAGE_MCAP_PUBLIC ROSBAG2_STORAGE_MCAP_IMPORT
   #endif
-  #define ROSBAG2_STORAGE_MCAP_PUBLIC_TYPE ROSBAG2_STORAGE_MCAP_PUBLIC
-  #define ROSBAG2_STORAGE_MCAP_LOCAL
 #else
   #define ROSBAG2_STORAGE_MCAP_EXPORT __attribute__((visibility("default")))
   #define ROSBAG2_STORAGE_MCAP_IMPORT
   #if __GNUC__ >= 4
     #define ROSBAG2_STORAGE_MCAP_PUBLIC __attribute__((visibility("default")))
-    #define ROSBAG2_STORAGE_MCAP_LOCAL __attribute__((visibility("hidden")))
   #else
     #define ROSBAG2_STORAGE_MCAP_PUBLIC
-    #define ROSBAG2_STORAGE_MCAP_LOCAL
   #endif
-  #define ROSBAG2_STORAGE_MCAP_PUBLIC_TYPE
 #endif
 
 #endif  // ROSBAG2_STORAGE_MCAP__VISIBILITY_CONTROL_HPP_

--- a/rosbag2_storage_mcap/include/rosbag2_storage_mcap/visibility_control.hpp
+++ b/rosbag2_storage_mcap/include/rosbag2_storage_mcap/visibility_control.hpp
@@ -1,0 +1,56 @@
+// Copyright 2022 Foxglove Technologies Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* This header must be included by all rclcpp headers which declare symbols
+ * which are defined in the rclcpp library. When not building the rclcpp
+ * library, i.e. when using the headers in other package's code, the contents
+ * of this header change the visibility of certain symbols which the rclcpp
+ * library cannot have, but the consuming code must have inorder to link.
+ */
+
+#ifndef ROSBAG2_STORAGE_MCAP__VISIBILITY_CONTROL_HPP_
+#define ROSBAG2_STORAGE_MCAP__VISIBILITY_CONTROL_HPP_
+
+// This logic was borrowed (then namespaced) from the examples on the gcc wiki:
+//     https://gcc.gnu.org/wiki/Visibility
+
+#if defined _WIN32 || defined __CYGWIN__
+  #ifdef __GNUC__
+    #define ROSBAG2_STORAGE_MCAP_EXPORT __attribute__((dllexport))
+    #define ROSBAG2_STORAGE_MCAP_IMPORT __attribute__((dllimport))
+  #else
+    #define ROSBAG2_STORAGE_MCAP_EXPORT __declspec(dllexport)
+    #define ROSBAG2_STORAGE_MCAP_IMPORT __declspec(dllimport)
+  #endif
+  #ifdef ROSBAG2_STORAGE_MCAP_BUILDING_DLL
+    #define ROSBAG2_STORAGE_MCAP_PUBLIC ROSBAG2_STORAGE_MCAP_EXPORT
+  #else
+    #define ROSBAG2_STORAGE_MCAP_PUBLIC ROSBAG2_STORAGE_MCAP_IMPORT
+  #endif
+  #define ROSBAG2_STORAGE_MCAP_PUBLIC_TYPE ROSBAG2_STORAGE_MCAP_PUBLIC
+  #define ROSBAG2_STORAGE_MCAP_LOCAL
+#else
+  #define ROSBAG2_STORAGE_MCAP_EXPORT __attribute__((visibility("default")))
+  #define ROSBAG2_STORAGE_MCAP_IMPORT
+  #if __GNUC__ >= 4
+    #define ROSBAG2_STORAGE_MCAP_PUBLIC __attribute__((visibility("default")))
+    #define ROSBAG2_STORAGE_MCAP_LOCAL __attribute__((visibility("hidden")))
+  #else
+    #define ROSBAG2_STORAGE_MCAP_PUBLIC
+    #define ROSBAG2_STORAGE_MCAP_LOCAL
+  #endif
+  #define ROSBAG2_STORAGE_MCAP_PUBLIC_TYPE
+#endif
+
+#endif  // ROSBAG2_STORAGE_MCAP__VISIBILITY_CONTROL_HPP_

--- a/rosbag2_storage_mcap/src/mcap_storage.cpp
+++ b/rosbag2_storage_mcap/src/mcap_storage.cpp
@@ -19,21 +19,21 @@
 #include "rosbag2_storage_mcap/message_definition_cache.hpp"
 
 #ifdef ROSBAG2_STORAGE_MCAP_HAS_YAML_HPP
-#include "rosbag2_storage/yaml.hpp"
+  #include "rosbag2_storage/yaml.hpp"
 #else
-// COMPATIBILITY(foxy, galactic) - this block is available in rosbag2_storage/yaml.hpp in H
-#ifdef _WIN32
-// This is necessary because of a bug in yaml-cpp's cmake
-#define YAML_CPP_DLL
-// This is necessary because yaml-cpp does not always use dllimport/dllexport consistently
-#pragma warning(push)
-#pragma warning(disable : 4251)
-#pragma warning(disable : 4275)
-#endif
-#include "yaml-cpp/yaml.h"
-#ifdef _WIN32
-#pragma warning(pop)
-#endif
+  // COMPATIBILITY(foxy, galactic) - this block is available in rosbag2_storage/yaml.hpp in H
+  #ifdef _WIN32
+    // This is necessary because of a bug in yaml-cpp's cmake
+    #define YAML_CPP_DLL
+    // This is necessary because yaml-cpp does not always use dllimport/dllexport consistently
+    #pragma warning(push)
+    #pragma warning(disable : 4251)
+    #pragma warning(disable : 4275)
+  #endif
+  #include "yaml-cpp/yaml.h"
+  #ifdef _WIN32
+    #pragma warning(pop)
+  #endif
 #endif
 
 #include <mcap/mcap.hpp>
@@ -47,7 +47,7 @@
 #include <utility>
 #include <vector>
 #ifdef ROSBAG2_STORAGE_MCAP_HAS_STORAGE_FILTER_TOPIC_REGEX
-#include <regex>
+  #include <regex>
 #endif
 
 #define DECLARE_YAML_VALUE_MAP(KEY_TYPE, VALUE_TYPE, ...)                   \

--- a/rosbag2_storage_mcap/src/message_definition_cache.cpp
+++ b/rosbag2_storage_mcap/src/message_definition_cache.cpp
@@ -20,6 +20,7 @@
 #include <rcutils/logging_macros.h>
 
 #include <fstream>
+#include <functional>
 #include <optional>
 #include <regex>
 #include <set>

--- a/rosbag2_storage_mcap/test/rosbag2_storage_mcap/test_mcap_storage.cpp
+++ b/rosbag2_storage_mcap/test/rosbag2_storage_mcap/test_mcap_storage.cpp
@@ -20,10 +20,10 @@
 #include "rosbag2_cpp/writer.hpp"
 #include "rosbag2_cpp/writers/sequential_writer.hpp"
 #ifdef ROSBAG2_STORAGE_MCAP_HAS_STORAGE_OPTIONS
-#include "rosbag2_storage/storage_options.hpp"
+  #include "rosbag2_storage/storage_options.hpp"
 using StorageOptions = rosbag2_storage::StorageOptions;
 #else
-#include "rosbag2_cpp/storage_options.hpp"
+  #include "rosbag2_cpp/storage_options.hpp"
 using StorageOptions = rosbag2_cpp::StorageOptions;
 #endif
 #include "rosbag2_test_common/temporary_directory_fixture.hpp"
@@ -128,9 +128,9 @@ TEST_F(TemporaryDirectoryFixture, can_write_mcap_with_zstd_configured_from_yaml)
     msg.data = message_data;
 
     rosbag2_cpp::Writer writer{std::make_unique<rosbag2_cpp::writers::SequentialWriter>()};
-#ifndef ROSBAG2_STORAGE_MCAP_WRITER_CREATES_DIRECTORY
+  #ifndef ROSBAG2_STORAGE_MCAP_WRITER_CREATES_DIRECTORY
     rcpputils::fs::create_directories(uri);
-#endif
+  #endif
     writer.open(options, rosbag2_cpp::ConverterOptions{});
     writer.create_topic(topic_metadata);
 

